railties/lib/rails/generators/rails/app/templates/Gemfile

```diff
@@ -1,6 +1,6 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
-ruby <%= "'#{RUBY_VERSION}'" %>
+ruby <%= "'#{RUBY_VERSION}'" -%>
```

Comment:
The empty line following `<% if gem.comment -%>` is required when commenting in a loop. 
If there is no empty line, the comment will collapse as shown below. 

``` 
# Use sqlite3 as the database for Active Record
gem 'sqlite3'# Use Puma as the app server
gem 'puma', '~> 3.7'# Use SCSS for stylesheets
gem 'sass-rails', '~> 5.0'# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'# See https://github.com/rails/execjs#readme for more supported runtimes
``` 

So, instead of remove this emptyline, I fixed by adding a empty line to the beginning.